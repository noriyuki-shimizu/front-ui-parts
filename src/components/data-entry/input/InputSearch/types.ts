/** Props */
export interface Props {
  /** placeholder */
  placeholder?: string
  /** size */
  size?: 'small' | 'medium' | 'large'
}
