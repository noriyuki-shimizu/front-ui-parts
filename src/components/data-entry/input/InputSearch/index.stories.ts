import type { ArgTypeObject, Meta, StoryObj } from '@storybook/vue3'
import { ref } from 'vue'
import InputSearch from './index.vue'

// TODO: Watch: https://github.com/storybookjs/storybook/issues/24238
const InputSearchComponent = InputSearch as Record<string, unknown>

const meta = {
  title: 'Data Entry/InputSearch',
  component: InputSearchComponent,
  tags: ['autodocs']
} satisfies Meta<typeof InputSearchComponent> & ArgTypeObject // TODO: Watch: https://github.com/storybookjs/storybook/issues/23352

export default meta

type Story = StoryObj<typeof meta>

const template: Story = {
  render: (args) => ({
    components: { InputSearch: InputSearchComponent },
    setup() {
      const inputText = ref<string>('')
      return { args, inputText }
    },
    template: `
      <InputSearch
        v-model="inputText"
        :placeholder="args.placeholder"
        :size="args.size"
      />
    `
  })
}

export const Default: Story = {
  ...template,
  args: {
    placeholder: 'タイトル検索',
    size: 'medium'
  }
}

export const Small: Story = {
  ...template,
  args: {
    placeholder: 'タイトル検索',
    size: 'small'
  }
}

export const Large: Story = {
  ...template,
  args: {
    placeholder: 'タイトル検索',
    size: 'large'
  }
}
