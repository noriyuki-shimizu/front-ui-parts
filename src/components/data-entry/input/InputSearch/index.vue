<script setup lang="ts">
import type { Props } from './types'
import SearchIconSvg from '@/assets/svg/search-icon.svg?component'

/** Props */
const props = withDefaults(defineProps<Props>(), {
  placeholder: 'タイトル検索',
  size: 'medium',
  inputAriaLabel: '検索フィールド',
  buttonAriaLabel: '検索ボタン'
})

/** 入力値 */
const model = defineModel<string | null>({ required: true })
</script>

<template>
  <div :class="$style['input-search']">
    <input
      v-model="model"
      type="text"
      :placeholder="props.placeholder"
      :class="[$style['input-search__input'], $style[`input-search__input--${$props.size}`]]"
      :aria-label="props.inputAriaLabel"
    />
    <button type="submit" title="search" :class="$style['input-search__button']" :aria-label="props.buttonAriaLabel">
      <SearchIconSvg :class="$style['input-search__icon']" />
    </button>
  </div>
</template>

<style module lang="scss">
@use './style.scss';
</style>
