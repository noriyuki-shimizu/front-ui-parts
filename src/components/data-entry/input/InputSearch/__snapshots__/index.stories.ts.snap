// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Entry/InputSearch Default smoke-test 1`] = `
<div class="_input-search_1w78h_325">
  <input type="text"
         placeholder="タイトル検索"
         class="_input-search__input_1w78h_328 _input-search__input--medium_1w78h_337"
  >
  <button type="submit"
          title="search"
          class="_input-search__button_1w78h_343"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="16"
         height="16"
         viewbox="0 0 512 512"
         class="_input-search__icon_1w78h_356"
    >
      <path fill="currentColor"
            d="M416 208c0 45.9-14.9 88.3-40 122.7l126.6 126.7c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208M208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288"
      >
      </path>
    </svg>
  </button>
</div>
`;

exports[`Data Entry/InputSearch Large smoke-test 1`] = `
<div class="_input-search_1w78h_325">
  <input type="text"
         placeholder="タイトル検索"
         class="_input-search__input_1w78h_328 _input-search__input--large_1w78h_340"
  >
  <button type="submit"
          title="search"
          class="_input-search__button_1w78h_343"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="16"
         height="16"
         viewbox="0 0 512 512"
         class="_input-search__icon_1w78h_356"
    >
      <path fill="currentColor"
            d="M416 208c0 45.9-14.9 88.3-40 122.7l126.6 126.7c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208M208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288"
      >
      </path>
    </svg>
  </button>
</div>
`;

exports[`Data Entry/InputSearch Small smoke-test 1`] = `
<div class="_input-search_1w78h_325">
  <input type="text"
         placeholder="タイトル検索"
         class="_input-search__input_1w78h_328 _input-search__input--small_1w78h_334"
  >
  <button type="submit"
          title="search"
          class="_input-search__button_1w78h_343"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="16"
         height="16"
         viewbox="0 0 512 512"
         class="_input-search__icon_1w78h_356"
    >
      <path fill="currentColor"
            d="M416 208c0 45.9-14.9 88.3-40 122.7l126.6 126.7c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208M208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288"
      >
      </path>
    </svg>
  </button>
</div>
`;
